cmd/korg/utils.go

```diff
@@ -0,0 +1,183 @@
+		} else {
+			valid = append(valid, org)
+		}
+	}
+
+	if len(invalid) > 0 {
+		invalidPresent = true
+	}
+
+	return
+}
+
+func readConfig(path string) (*org.Config, error) {
+	contents, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read file at %s: %s", path, err)
+	}
+
+	config := org.Config{}
+	if err := yaml.Unmarshal(contents, &config); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal config from %s: %s", path, err)
+	}
+
+	return &config, nil
+}
+
+func saveConfig(path string, config *org.Config) error {
+	b, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("unable to marshal config: %s", err)
+	}
+
+	info, err := os.Stat(path)
+	if err != nil {
+		return fmt.Errorf("unable to fetch info for %s: %s", path, err)
+	}
+
+	if err := os.WriteFile(path, b, info.Mode()); err != nil {
+		return fmt.Errorf("unable to write to %s: %s", path, err)
+	}
+	return nil
+}
+
+func commitChanges(repoRoot string, configsModified []string, message string) error {
+	r, err := git.PlainOpen(repoRoot)
+	if err != nil {
+		return fmt.Errorf("unable to open repository: %s", err)
+	}
+
+	w, err := r.Worktree()
+	if err != nil {
+		return fmt.Errorf("unable to fetch worktree: %s", err)
+	}
+
+	for _, configModified := range configsModified {
+		_, err := w.Add(configModified)
+		if err != nil {
+			return fmt.Errorf("unable to stage changes: %s", err)
+		}
+	}
+
+	commit, err := w.Commit(message, &git.CommitOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to commit changes: %s", err)
+	}
+
+	_, err = r.CommitObject(commit)
+	if err != nil {
+		return fmt.Errorf("unable to write commit object to repo: %s", err)
+	}
+
+	return nil
+}
+
+// this is an custom implementation which sorts a string slice
+// with a case agnostic heuristic. the default sorting algorithm
+// in Go doesn't ignore case, whereas we do.
+type caseAgnostic []string
+
+func (s caseAgnostic) Len() int {
+	return len(s)
+}
+
+func (s caseAgnostic) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func (s caseAgnostic) Less(i, j int) bool {
+	li := strings.ToLower(s[i])
+	lj := strings.ToLower(s[j])
+
+	return sort.StringsAreSorted([]string{li, lj})
+}
+
+func caseAgnosticSort(arr []string) {
+	sort.Sort(caseAgnostic(arr))
+}
+
+func parseTeam(team string) (string, string, string, error) {
+	parts := strings.Split(team, "/")
+	if len(parts) < 2 || len(parts) > 3 {
+		return "", "", "", fmt.Errorf("invalid team: %s", team)
+	}
+
+	org := parts[0]
+	if !stringInSlice(validOrgs, org) {
+		return "", "", "", fmt.Errorf("invalid team: %s", team)
+	}
+
+	group := ""
+	if len(parts) == 2 {
+		group = ""
+		team = parts[1]
+	} else {
+		group = parts[1]
+		team = parts[2]
+	}
+
+	return org, group, team, nil
+
+}
+
+func userInOrg(username string, org string, options Options) bool {
+	configPath := filepath.Join(options.RepoRoot, fmt.Sprintf(orgConfigPathFormat, org))
+	config, err := readConfig(configPath)
```

Comment:
Looks like we need to do the "is user in the org" check for both adding a user to an org and a team. Can we instead read the config once before we call `AddMemberToOrgs` and `AddMemberToTeams` instead of calling it twice?

Reply:
I agree to the suggestion. Reading the config at the start of the command and reusing it makes more sense.

Since the team implementation is deferred to later, we don't need to cache the config in memory for now.